static/js/containers/ChannelPage.js

```diff
@@ -101,20 +61,37 @@ class ChannelPage extends React.Component {
       )
     }
   }
+}
 
-  render() {
-    const { channels, postsForChannel } = this.props
-    return <Loading restStates={[channels, postsForChannel]} renderContents={this.renderContents} />
-  }
+const shouldUpdateRequirements = R.complement(R.eqProps("channelName"))
```

Comment:
Just to clarify - `R.complement` and `R.not` aren't equivalent,

```
R.not :: Boolean -> Boolean
R.complement :: (Any -> Boolean) -> Boolean
```

`R.complement` takes a function as an argument, where `R.not` takes a boolean value. To rewrite with `R.not` you'd need to add a wrapping call to `R.compose`.